source/developers-guide/legacy-plugin-system/index.md

```diff
@@ -0,0 +1,820 @@
+            image = me.getConfigValue('vimeo_video_thumbnail'),
+            style;
+
+        if (Ext.isDefined(image)) {
+            style = Ext.String.format('background-image: url([0]);', image);
+
+            preview = Ext.String.format('<div class="x-emotion-banner-element-preview" style="[0]"></div>', style);
+        }
+
+        return preview;
+    }
+});
+//{/block}
+```
+
+## Add a new payment method
+Use the "createPayment" method to add payment methods to the database inside plugin installations.
+
+```php
+public function install(InstallContext $context)
+{
+    $this->createPayment(
+        array(
+            'name' => 'payment name',
+            'description' => 'Pay by the new Payment',
+            'action' => 'payment_paymentName',
+            'active' => 0,
+            'position' => 0,
+            'additionalDescription' => 'Lorem ipsum ... ',
+        )
+    );
+}
+```
+
+### Plugin Configuration / Forms
+
+Backend plugin configuration can be extended by the usage of `$this->Form()->setElement()` in the install method of the plugin bootstrap.php. 
+
+```php
+$form = $this->Form();
+
+// Now we can use this instance of Shopware\Models\Config\Form in $form to add elements to it via `setElement().
+
+$form->setElement(
+    
+    'color', 
+    'yourSetting',
+    [
+        'label' => 'Your label',
+        'description' => Lorem ipsum ...',
+        'value' => '#ffffff',
+        'scope' => Element::SCOPE_SHOP,
+    ]
+);
+```
+
+A textfield would be defined as followed:
+```php
+public function createConfiguration()
+{
+    $form = $this->Form();
+ 
+    $form->setElement('text', 'simpleTextField',
+        array(
+            'label' => 'Text',
+            'value' => 'Preselection',
+            'scope' => Shopware\Models\Config\Element::SCOPE_SHOP,
+            'description' => 'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut',
+            'required' => true
+        )
+    ); 
+}
+```
+
+#### Options parameter
+The options parameter of the setElement function allows to set several configurations on the form element.
+* **Label**
+The label parameter allows to create a simple descriptional label for the form element.
+* **Value**
+The value parameter stands for the default value of the field if this hasn´t been edited yet. It will directly be shown in the configuration element.
+* **Scope**
+With help of the scope parameter it is possible to generate subshop specific configurations. Leaving this option out results in a configuration option that applies for all subshops.
+* **Description**
+The description parameter allows to provide a more detailed description of the configuration element.
+* **Required**
+The required parameter specifies whether the configuration item is mandatory or not.
+
+#### Possible elements: 
+* color
+* date
+* datetime 
+* html 
+* interval 
+* mediaselection, 
+* number, 
+* select, 
+* text, 
+* textarea 
+* time
+
```

Comment:
the translation of config elements should also be shown

Reply:
D